# MP2 Part 1 (jtyuill)

In [1]:
%pip install -q python-woc pandas

Note: you may need to restart the kernel to use updated packages.


In [2]:
import os
import time
import pandas as pd
from woc.remote import WocMapsRemote

woc = WocMapsRemote(base_url='https://worldofcode.org/api/')
assigned = pd.read_csv('net2prj.csv')
projects = list(assigned[assigned['netID'] == 'jtyuill']['WoC'])

if os.path.exists('jtyuill_project_summary.csv'):
    # already downloaded, reuse it instead of pulling every commit again
    summary = pd.read_csv('jtyuill_project_summary.csv', sep=';')
else:
    rows = []
    for project in projects:
        shas = woc.get_values('p2c', project)
        for start in range(0, len(shas), 10):
            results, errors = woc.get_values_many('commit.tch', shas[start:start + 10])
            for sha, value in results.items():
                tree, parents, author, committer, message = value[0]
                rows.append({
                    'project_wocid': project,
                    'commit_sha1': sha,
                    'author': author[0],
                    'time': int(author[1]),
                    'commit message': message,
                })
            time.sleep(1)
    summary = pd.DataFrame(rows)
    summary.to_csv('jtyuill_project_summary.csv', sep=';', index=False)

summary['project_wocid'].value_counts()

project_wocid
tensorflow_tfjs                   31325
ifcopenshell_ifcopenshell         25237
opendatacube_datacube-explorer     6011
lastools_lastools                  2712
choderalab_pymbar                  1492
jakobrunge_tigramite               1379
clue-developers_clue                842
johannfaouzi_pyts                   791
kvos_coastsat                       642
zalo_cascadestudio                  361
Name: count, dtype: int64

In [3]:
stats = []
for project in projects:
    commits = summary[summary['project_wocid'] == project]
    stats.append({
        'project': project,
        'commits': len(commits),
        'authors': commits['author'].nunique(),
        'min time': commits['time'].min(),
        'max time': commits['time'].max(),
    })
pd.DataFrame(stats)

,project,commits,authors,min time,max time
0,ifcopenshell_ifcopenshell,25237,428,1304852672,1762623536
1,lastools_lastools,2712,87,1413703399,1761411541
2,opendatacube_datacube-explorer,6011,53,1468452526,1762320537
3,choderalab_pymbar,1492,61,1368225944,1771982683
4,tensorflow_tfjs,31325,705,1501782313,1776389827
5,kvos_coastsat,642,30,1538116640,1768665837
6,jakobrunge_tigramite,1379,56,1486320043,1762336405
7,clue-developers_clue,842,17,1567025867,1726151591
8,zalo_cascadestudio,361,14,1595495597,1741942474
9,johannfaouzi_pyts,791,30,1501756392,1750449970


## GitHub

| Project | Stars | Forks | Last commit |
|:--|--:|--:|:--|
| ifcopenshell_ifcopenshell | 2808 | 973 | 2026-09-28 |
| lastools_lastools | 1070 | 405 | 2026-09-21 |
| opendatacube_datacube-explorer | 64 | 35 | 2026-09-25 |
| choderalab_pymbar | 318 | 101 | 2026-02-12 |
| tensorflow_tfjs | 19146 | 2029 | 2026-04-06 |
| kvos_coastsat | 898 | 309 | 2026-05-29 |
| jakobrunge_tigramite | 1726 | 323 | 2026-01-14 |
| clue-developers_clue | 14 | 11 | 2024-09-12 |
| zalo_cascadestudio | 1484 | 181 | 2026-06-16 |
| johannfaouzi_pyts | 1880 | 179 | 2025-06-18 |

## world of code

| Project | Commits | Authors | Min time | Max time |
|:--|--:|--:|:--|:--|
| ifcopenshell_ifcopenshell | 25237 | 428 | 1304852672 (2011-05-08) | 1762623536 (2025-11-08) |
| lastools_lastools | 2712 | 87 | 1413703399 (2014-10-19) | 1761411541 (2025-10-25) |
| opendatacube_datacube-explorer | 6011 | 53 | 1468452526 (2016-07-13) | 1762320537 (2025-11-05) |
| choderalab_pymbar | 1492 | 61 | 1368225944 (2013-05-10) | 1771982683 (2026-02-25) |
| tensorflow_tfjs | 31325 | 705 | 1501782313 (2017-08-03) | 1776389827 (2026-04-17) |
| kvos_coastsat | 642 | 30 | 1538116640 (2018-09-28) | 1768665837 (2026-01-17) |
| jakobrunge_tigramite | 1379 | 56 | 1486320043 (2017-02-05) | 1762336405 (2025-11-05) |
| clue-developers_clue | 842 | 17 | 1567025867 (2019-08-28) | 1726151591 (2024-09-12) |
| zalo_cascadestudio | 361 | 14 | 1595495597 (2020-07-23) | 1741942474 (2025-03-14) |
| johannfaouzi_pyts | 791 | 30 | 1501756392 (2017-08-03) | 1750449970 (2025-06-20) |